File: 3503.longest-palindrome-after-substring-concatenation-i.java
#
# @lc app=leetcode id=3503 lang=java
#
# [3503] Longest Palindrome After Substring Concatenation I
#

# @lc code=start
class Solution {
    public int longestPalindrome(String s, String t) {
        int maxLen = 0;
        int n = s.length(), m = t.length();
        for (int i = 0; i <= n; ++i) {
            for (int j = i; j <= n; ++j) {
                String ss = s.substring(i, j);
                for (int k = 0; k <= m; ++k) {
                    for (int l = k; l <= m; ++l) {
                        String tt = t.substring(k, l);
                        String combined = ss + tt;
                        if (combined.length() > 0 && isPalindrome(combined)) {
                            maxLen = Math.max(maxLen, combined.length());
                        }
                    }
                }
            }
        }
        return maxLen;
    }
    private boolean isPalindrome(String str) {
        int left = 0, right = str.length() - 1;
        while (left < right) {
            if (str.charAt(left) != str.charAt(right)) return false;
            left++; right--;
        }
        return true;
    }
}
# @lc code=end

File: 3503.longest-palindrome-after-substring-concatenation-i.py
#
# @lc app=leetcode id=3503 lang=python3
#
# [3503] Longest Palindrome After Substring Concatenation I
#

# @lc code=start
class Solution:
    def longestPalindrome(self, s: str, t: str) -> int:
        def is_palindrome(x):
            return x == x[::-1]
        max_len = 0
        n, m = len(s), len(t)
        # Try all substrings from s
        for i in range(n + 1):
            for j in range(i, n + 1):
                sub_s = s[i:j]
                # Try all substrings from t
                for k in range(m + 1):
                    for l in range(k, m + 1):
                        sub_t = t[k:l]
                        combined = sub_s + sub_t
                        if combined and is_palindrome(combined):
                            max_len = max(max_len, len(combined))
        return max_len
# @lc code=end